tree_sitter/binding.c

```diff
@@ -134,6 +134,45 @@ static PyObject *node_chield_by_field_name(Node *self, PyObject *args) {
   return node_new_internal(child, self->tree);
 }
 
+static PyObject *node_children_by_field_id_internal(Node *self, TSFieldId field_id) {
+  PyObject *result = PyList_New(0);
+  TSTreeCursor cursor = ts_tree_cursor_new(self->node);
```

Comment:
I think that this `TSTreeCursor` will be leaked. You would have to destroy it with `ts_tree_cursor_delete`. But actually, you should probably use the same approach as `node_get_children` does below, where it reused a global `TSTreeCursor` called `default_cursor` (to avoid unnecessary allocations).

Reply:
Indeed, updated code to use `default_cursor`.